src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -94,13 +96,7 @@ public void Start<TContext>(IHttpApplication<TContext> application)
                 engine.Start(threadCount);
                 var atLeastOneListener = false;
 
-                var addressesFeature = Features.Get<IServerAddressesFeature>();
-                if (addressesFeature == null)
-                {
-                    throw new InvalidOperationException($"{nameof(IServerAddressesFeature)} is missing.");
-                }
-
-                foreach (var address in addressesFeature.Addresses)
+                foreach (var address in _serverAddresses.Addresses)
```

Comment:
I consciously prevented that. The server owns the FeatureCollection and may map the IServerAddressesFeature to some internal structure. We don't want apps to replace it and have it work on some servers and not others.